### PR TITLE
[CT-853] Generate order replacement protos

### PR DESCRIPTION
**Pull request by chenyaoy**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced `StatefulOrderEventV1` with enhanced functionalities.
  - Added `OrderReplaceV1` with new encoding and decoding capabilities.

- **Improvements**
  - Renamed several order placement status identifiers for better clarity and consistency.
  - Enhanced `OrderPlaceV1` and `OrderReplaceV1` with updated status types for improved order management.

### CODE REVIEW

File: proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto [hunk 1]
```diff
@@ -88,13 +88,25 @@ message OrderUpdateV1 {
   uint64 total_filled_quantums = 2;
 }
 
+// OrderReplace messages contain the replacement order.
+message OrderReplaceV1 {
+  dydxprotocol.indexer.protocol.v1.IndexerOrder order = 1;
+  dydxprotocol.indexer.off_chain_updates.OrderPlaceV1.OrderPlacementStatus
```
Comment (dydxwill) [inline proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto:94]: Do we want new enum values for replacement? Feel free to handle in different PR.

File: proto/dydxprotocol/indexer/events/events.proto [hunk 2]
```diff
@@ -249,6 +249,7 @@ message StatefulOrderEventV1 {
     ConditionalOrderPlacementV1 conditional_order_placement = 5;
     ConditionalOrderTriggeredV1 conditional_order_triggered = 6;
     LongTermOrderPlacementV1 long_term_order_placement = 7;
+    StatefulOrderReplacementV1 order_replace = 8;
```
Comment (dydxwill) [inline proto/dydxprotocol/indexer/events/events.proto:252]: LongTermOrderReplacementV1?

Comment (chenyaoy) [reply proto/dydxprotocol/indexer/events/events.proto]: changed!

File: proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto (hunk 1 above)
Comment (chenyaoy) [reply proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto:94]: I'm not sure, do you see a use case for new values?

Comment (dydxwill) [reply proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto:94]: to distinguish between placed vs replaced orders.

**Review by jayy04 — APPROVED** [review]

**Review by dydxwill — APPROVED** [review]

File: proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto (hunk 1 above)
Comment (chenyaoy) [reply proto/dydxprotocol/indexer/off_chain_updates/off_chain_updates.proto:94]: Discussed offline, will leave this for now. We can distinguish based on message type

**Comment by chenyaoy** [comment]
@Mergifyio backport release/indexer/v5.x

**Pull request merged**